File: lib/widgets/single_course/single_course_widget_state.dart
import 'package:flutter/cupertino.dart';
import 'package:swustmeow/widgets/entities/single_course.dart';

class SingleCourseWidgetState {
  final success = ValueNotifier(true);
  final lastUpdateTimestamp =
      ValueNotifier(DateTime.now().millisecondsSinceEpoch);
  final current = ValueNotifier<SingleCourse?>(null);
  final next = ValueNotifier<SingleCourse?>(null);
  final weekNum = ValueNotifier<int>(0);

  void clear() {
    current.value = null;
    next.value = null;
    weekNum.value = 0;
  }
}

File: android/app/src/main/kotlin/store/swust/swustmeow/widgets/single_course/SingleCourseWidgetState.kt
package store.swust.swustmeow.widgets.single_course

import store.swust.swustmeow.entities.SingleCourse

data class SingleCourseWidgetState(
    val success: Boolean = false,
    val lastUpdateTimestamp: Long = 0,
    val currentCourse: SingleCourse? = null,
    val nextCourse: SingleCourse? = null,
    val weekNum: Int = 0
)
